src/main/java/io/pinecone/clients/Pinecone.java

```diff
@@ -99,6 +104,301 @@ public IndexModel createServerlessIndex(String indexName, String metric, int dim
         return indexModel;
     }
 
+    // Minimal
+    public IndexModel createPodsIndex(String indexName, Integer dimension, String environment, String podType) {
```

Comment:
We have redundant checks in the method. This method is checking for null and emptiness, and at line#129, we are calling createPodsIndex(indexName, dimension, environment, null, podType, null, null, null, null, null) which further checks for the null and emptiness.